File: Pkdd.Models/Person/Person.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Pkdd.Abstractions.Entity;
using Pkdd.Models.Persons.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Pkdd.Models.Persons
{
    public class Person : EntityBase<Person>
    {
        [Required, StringLength(200)]
        public string Name { get; set; }

        public Sexes Sex { get; set; }

        public DateTime Birthday { get; set; }

        /// <summary>
        /// Перечисление текущих мест работы и соответсвующих должностей
        /// </summary>
        [StringLength(500)]
        public string Position { get; set; }

        public Priority Priority { get; set; }

        public string PhotoUrl { get; set; }

        public bool IsPublished { get; set; }

        public BaseBioBlock BioBlock { get; set; }

        [NotMapped]
        public int ResultsCount { get; set; }

        public int Views { get; set; }

        public override Person Update(Person entity)
        {
            Name = entity.Name;
            Sex = entity.Sex;
            Birthday = entity.Birthday;
            Position = entity.Position;
            IsPublished = entity.IsPublished;
            PhotoUrl = entity.PhotoUrl;
            this.MarkUpdated();
            return this;
        }

    }
}

File: Pkdd/pkdd-ng-client/src/app/models/entities/person.ts
import { IEntity } from './interfaces/iEntity';
import { Sexes } from './enums/sexes';
import { TimeTrack } from '../common/time-track';
import { BaseBioBlockBackend, BaseBioBlock } from './base-bio-block';
import { Priority } from './enums/priority';

abstract class AbstractPerson implements IEntity {

    public id: number;
    public isDeleted: boolean;
    public timeTrack: TimeTrack;
    public name: string;
    public sex: Sexes;
    public birthday: Date;
    public position: string;
    public photoUrl: string;
    public isPublished: boolean;
    public priority: Priority;

    constructor(person: AbstractPerson) {
        this.id = person.id;
        this.isDeleted = person.isDeleted;
        this.timeTrack = person.timeTrack;
        this.name = person.name;
        this.sex = person.sex;
        this.birthday = person.birthday;
        this.position = person.position;
        this.photoUrl = person.photoUrl;
        this.isPublished = person.isPublished;
        this.priority = person.priority;
    }
}

export class PersonBackend extends AbstractPerson {
    bioBlock: BaseBioBlockBackend;

    constructor(person: AbstractPerson, bioBlock: BaseBioBlockBackend) {
        super(person);
        this.bioBlock = bioBlock;
    }
}

export class Person extends AbstractPerson {
    bioBlock: BaseBioBlock;

    constructor(person: AbstractPerson, bioBlock: BaseBioBlock) {
        super(person);
        this.bioBlock = bioBlock;
    }
}
